dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -963,7 +975,20 @@ private static void checkHasSourceFiles(WorkflowVersion workflowVersion) {
      */
     public static boolean canAutomaticallyCreateDockstoreOwnedDoi(Workflow workflow, WorkflowVersion workflowVersion) {
         final boolean validPublishedTag = workflow.getIsPublished() && workflowVersion.isValid() && workflowVersion.getReferenceType() == ReferenceType.TAG;
-        return validPublishedTag && !hasExistingDOIForWorkflowVersion(workflowVersion, DoiInitiator.DOCKSTORE);
+        return validPublishedTag && isAutoCreateDockstoreOwnedDoiAllowed(workflow) && !hasExistingDOIForWorkflowVersion(workflowVersion, DoiInitiator.DOCKSTORE);
+    }
+
+    /**
+     * Indicates whether auto-creating a Dockstore owned DOI for a workflow is allowed, based on the configuration file
+     * property <code>autoGeneratedDoiAllowList</code> value. If the value is null or empty, then creating a DOI is allowed for all workflows,
+     * otherwise it's allowed for organizations or organization/repository values in the configuration value.
+     * @param workflow
+     * @return
+     */
+    public static boolean isAutoCreateDockstoreOwnedDoiAllowed(Workflow workflow) {
+        return autoGeneratedDoiAllowList.isEmpty()
```

Comment:
If the goal is to only generate DOIs for whitelisted orgs/repos, this line makes me a little worried, because if we mess up the configuration file, we might inadvertently generate them for everyone.

Reply:
> If the goal is to only generate DOIs for whitelisted orgs/repos, this code makes me a little worried, because if we mess up the configuration file and leave the line blank, we might inadvertently generate them for everyone.

This was my intent; I think our goal is to turn it on for a few orgs at first, then if all goes well, turn it on for everybody.